### PR TITLE
Adapt convert_digit_char for the case without radix argument

### PR DESCRIPTION
**Pull request by romainbrenguier**
This corrects issues with the `getNumericValue` function (see https://github.com/diffblue/test-gen/pull/851)

### CODE REVIEW

File: src/java_bytecode/character_refine_preprocess.cpp [hunk 1]
```diff
@@ -132,9 +132,13 @@ codet character_refine_preprocesst::convert_digit_char(
   conversion_inputt &target)
 {
   const code_function_callt &function_call=target;
-  assert(function_call.arguments().size()==2);
+  PRECONDITION(function_call.arguments().size()>=1);
```
Comment (thk123) [inline src/java_bytecode/character_refine_preprocess.cpp]: Is the case it should be either 1 or 2 - you might consider tightening this condition to `>= 1 && <= 2`

File: src/java_bytecode/character_refine_preprocess.cpp [hunk 2]
```diff
@@ -132,9 +132,13 @@ codet character_refine_preprocesst::convert_digit_char(
...
   const exprt &arg=function_call.arguments()[0];
-  const exprt &radix=function_call.arguments()[1];
+  // If there is no radix argument we set it to 36 which is the maximum possible
+  const exprt &radix=
+    function_call.arguments().size()>1?
+      function_call.arguments()[1]:
+      from_integer(36, arg.type());
```
Comment (thk123) [inline src/java_bytecode/character_refine_preprocess.cpp]: You might consider documenting this behaviour in the function comment (that it defaults to base 36).

**Comment by romainbrenguier** [comment]
@thk123 I added a test for this function: it causes a crash with the current version of CBMC but works correctly with this pull request.

File: regression/strings/CharacterGetNumericValue/test.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+public class test
+{
+    
+    public static void main(String[] args)
+    {
+	char c = 'a';
```
Comment (reuk) [inline regression/strings/CharacterGetNumericValue/test.java]: Indentation needs fixing but otherwise lgtm

**Pull request merged**